### PR TITLE
Add code for Interval List Intersection

### PR DESCRIPTION
**Pull request by ANUJAGRAWAL2000**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description
I have add the code, description and Input Output sample for the Interval List Intersection.
(Write your answer here.)

### Checklist

- [Yes ] I've read the contribution guidelines.
- [Yes ] I've checked the issue list before deciding what to submit.
- [Yes ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2751 is the issue related which this PR.
(Write your answer here.)

### CODE REVIEW

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Use proper code Format 
- Align your code properly
- Use proper naming conventions to define variables
- Add Time and Space Complexity at the end of the Code

**Comment by ANUJAGRAWAL2000** [comment]
I have updated the Interval List Intersection as align properly, format code, add Time complexity and space complexity, used proper naming conventions.

**Comment by ANUJAGRAWAL2000** [comment]
Kindly review @HarshCasper !!

File: C-Plus-Plus/cp/Interval_list_Intersection.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+the intersection of these two 
+interval lists.A closed interval 
+[a, b] (with a < b) denotes the set
+of real numbers x with a <= x <= b.
+The intersection of two closed 
+intervals is a set of real numbers
+that are either empty or represented
+as a closed interval. For example,
+the intersection of [1, 3] and 
+[2, 4] is [2, 3].
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Interval_list_Intersection.cpp]: Can you make this more compact and break it down into smaller sentences?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by ANUJAGRAWAL2000** [comment]
@HarshCasper I have make it more compact. Please look into it and merge.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**